README.md

```diff
@@ -31,38 +33,64 @@ const client = new SgidClient({
 })
 ```
 
+### Generate PKCE pair
+
+`generatePkcePair([length])`
+
+```typescript
+const { codeChallenge, codeVerifier } = generatePckePair()
+```
+
 ### Get Authorization URL
 
-`client.authorizationUrl(state, scope, [nonce], [redirectUri])`
+`async client.authorizationUrl(parameters)`
+
+- parameters: `<Object>`
+  - codeChallenge: `<string>`
+  - state: `<string>` (Optional)
+  - scope: `<string | string[]>` (Optional) **Default**: `myinfo.name openid`
+  - nonce: `<string> | <null>` (Optional) **Default**: Randomly generates a nonce if unspecified
+  - redirectUri: `<string>` (Optional) **Default**: Utilizes the `redirectUri` provided in the constructor
 
 ```typescript
-const { url } = client.authorizationUrl(
-  'state',
-  ['openid', 'myinfo.nric_number'], // or space-concatenated string
-  null, // defaults to randomly generated nonce if unspecified
-  'http://localhost:3000/other_callback', // overrides redirect uri
-)
+const { url } = client.authorizationUrl({
+  codeChallenge: 'zaqUHoBV3rnhBF2g0Gkz1qkpEZXHqi2OrPK1DqRi-Lk', // generated from the previous step
+  state: 'state', // optional as PKCE protects against CSRF
+  scope: ['openid', 'myinfo.name'], // or space-concatenated string
+  nonce: null, // defaults to randomly generated nonce if unspecified
+  redirectUri: 'http://localhost:3000/callback', // optional, to override redirect uri provided in constructor
+})
 ```
 
 ### Token exchange
 
-`async client.callback(code, [nonce], [redirectUri])`
+`async client.callback(parameters)`
+
+- parameters: `<Object>`
+  - code: `<string>`
+  - codeVerifier: `<string>`
+  - nonce: `<string>` (Optional)
+  - redirectUri: `<string>` (Optional)
 
 ```typescript
-const { sub, accessToken } = await client.callback(
-  'code', // auth code reuturned from redirect_url
-  null,
-  'http://localhost:3000/other_callback', // optional, unless overridden
-)
+const { sub, accessToken } = await client.callback({
```

Comment:
might be helpful to write a comment here mentioning what the `sub` is

Reply:
yup will do so